### PR TITLE
Add fence_chroma monitor action

### PR DESCRIPTION
**Pull request by utopiabound**
This cycles through availble agents, and returns "worst-case" return value.
Monitor return values (per API):
0 - everything is okay
1 - no connection to device
2 - fence device is reachable but in off state

Fixes Issue #75

Signed-off-by: Nathaniel Clark <nclark@whamcloud.com>

### CODE REVIEW

**Comment by jgrund** [comment]
This passed everything in SSI: https://build.whamcloud.com/job/integration-tests-shared-storage-configuration/1493/

File: chroma_agent/lib/fence_agents.py [hunk 1]
```diff
@@ -31,6 +31,10 @@ def list(self):
     def status(self):
         AgentShell.try_run(self.base_cmd + ["-n", self.plug, "-o", "status"])
 
+    def monitor(self):
```
Comment (johnsonw) [inline chroma_agent/lib/fence_agents.py]: Not a big deal, but I noticed the following pattern here:
```
["-n", self.plug, "-o", <x>]
```
You could probably make a function to generate the array; something like this:
```python
def get_plug_command(cmd):
   return self.base_cmd + ['-n', self.plug, "-o", cmd]
```
Then each of the statements could be updated to:
```python
AgentShell.try_run(get_plug_command(state))
...
AgentShell.try_run(get_plug_command("list-status"))
...
AgentShell.try_run(get_plug_command("status"))
...
```

File: tests/test_fencing.py [hunk 2]
```diff
@@ -267,6 +267,45 @@ def test_fence_agent_on_off(self):
...
+        self.reset_command_capture()
+        self.add_command(("cibadmin", "--query", "--local"))
+
+        # a node in standby should not be fenced
+        self.fake_node_attributes = self.fake_node_attributes.copy()
+        self.fake_node_attributes["standby"] = "on"
+
+        from chroma_agent.fence_chroma import main as agent_main
+
+        agent_args = ("-o", "off", "-n", self.fake_node_hostname)
```
Comment (johnsonw) [inline tests/test_fencing.py]: Any reason agent_args is defined? Looks like it's only being used once.

File: tests/test_fencing.py [hunk 3]
```diff
@@ -267,6 +267,45 @@ def test_fence_agent_on_off(self):
...
+
+
+class TestFenceAgentMonitor(FencingTestCase):
+    def setUp(self):
+        super(TestFenceAgentMonitor, self).setUp()
+
+        self.fake_node_attributes = {
+            "0_fence_agent": "fence_apc",
+            "0_fence_login": "admin",
+            "0_fence_password": "yourmom",
```
Comment (johnsonw) [inline tests/test_fencing.py]: Lol, we must come from the same generation :)

**Pull request merged**